server/src/main/java/org/apache/uniffle/server/ShuffleTaskInfo.java

```diff
@@ -106,13 +105,26 @@ public void setUser(String user) {
     this.user.set(user);
   }
 
-  public void setDataDistType(
-      ShuffleDataDistributionType dataDistType) {
-    this.dataDistType.set(dataDistType);
+  public int getMaxConcurrencyPerPartitionToWrite() {
+    return specification.getMaxConcurrencyPerPartitionToWrite();
   }
 
   public ShuffleDataDistributionType getDataDistType() {
-    return dataDistType.get();
+    return specification.getDistributionType();
+  }
+
+  public void setSpecification(ShuffleDataDistributionType type, int maxConcurrency) {
```

Comment:
Could we use `setSpecification(ShuffleSpecification spec)` here? We don't need synchronized, too.

Reply:
OK

Reply:
If so, we also should use the `AtomicReference`

Reply:
done